### ISSUE TITLE
Browse collection by grouping

### ISSUE THREAD

**Issue by KingDuckZ** [navidrome/navidrome#2045]
### Is your feature request related to a problem? Please describe.

I've added the `grouping` tag (from Picard) to my albums and I'd like to browse by that. That's especially useful for soundtracks, right now it's not possible to find everything that's related to a given game or show.

### Describe the solution you'd like

In the web ui there should be a way to browse by grouping if there is at least one album with that tag. For example, say you have the soundtracks of Wipeout, Wipeout 2097 and Wip3out. This is one of the examples where simply searching for "wipeout" won't work. By using the `grouping` tag I "tagged" the above albums with the "Wipeout" string. In Navidrome there should be a way to navigate to the list of groupings and browse albums that belong to the Wipeout group. Note that `grouping` can be a list, so for example if I tagged the albums as "Wipeout; PSX" I'd expect to be able to reach the above 3 albums by browsing into each grouping.

**Comment by deluan** [comment]
Yeah, we have it planned (see #1036), it is not there yet because it is a multi-valued field. Will keep this open to track work on this particular field.

**Comment by mstewart14** [comment]
I agree with the requestor, I use the grouping field as described in the original request or to group albums by "source" (music festival or friend who shared/suggested)

**Comment by KingDuckZ** [comment]
I had forgotten creating this ticket and opened a new one here https://github.com/navidrome/navidrome/discussions/2860 just the other day - same request, sorry for the duplicate!
Btw I agree with mstewart14 in that Navidrome already manages the genre array correctly.

**Comment by tfranken90** [comment]
This appears to have been implemented? At least the interface allows for filtering by grouping using the GRP1 tag. It may be worth better surfacing this feature as it effectively answers the following discussions... 

https://github.com/navidrome/navidrome/discussions/2851
https://github.com/navidrome/navidrome/discussions/2860
https://github.com/navidrome/navidrome/discussions/4603

In my library I map the  _releasegroup_series tag from MusicBrainz to GRP1 so I can browse releases by podcast series.

**Comment by mstewart14** [comment]
In a way the dynamic playlists solve this problem.  In that whatever hits you get on the tag are what appears and you can see what you got.  BUT what it doesn't do is reveal typos in my tags.  If I have marked many items (as I have) and was inconsistent (it's been a long time I've been collecting music) or just simply got tired and typo'd a tag then I may miss something... if the tags could be revealed the way genres are that would be slick.  
With the advent of dynamic playlists this is less exciting but for the reasons stated still a nice to have.

**Comment by KingDuckZ** [comment]
This ticket's age is approaching 4 years now. I'm not sure how much longer it will take for the API people to implement this but in the meantime, I was wondering, wouldn't it be possible for Navidrome to add an option to customise the genre tag name? I don't use the genre at all since they're usually all over the place anyways ("hip hop", "hip-hop", "unused", "unknown", "anime", "anime ost"... lol) so if it reported the groupings as the genre strings to client apps I would finally be able to browse by that

**Comment by mstewart14** [comment]
I'm using dynamic playlists with the grouping tag to find the stuff (or not) in my groupings.  I get that you want the API to return the grouping tags.  I don't think this fully works for your usecase but having the dynamic playlists be able to use the grouping tag is quite handy.

**Comment by Reinachan** [comment]
To add to this, I've currently been using genres to work around this in Jellyfin but want to migrate to using groupings instead once it becomes a viable alternative in Navidrome (and the apps I use). I basically want to use it similarly to OP, but for videogame music instead where an album is a videogame and a grouping is a game series or other kind of related collection.

But there's one thing I would like to be able to do on top of what is currently asked for. I'd like to be able to add images to represent these groupings. I understand that this violates the concept established when it comes to using only metadata, however, I'd argue that this is similar to playlists. You can make playlists with .m3u files and even when you export the playlists from Navidrome, the image does not carry with. Similarly I believe groupings is an area where it makes sense to have an image representation of what it is as I'm currently doing using a workaround in Jellyfin with genres in the example image below.

<img width="4237" height="2872" alt="Image" src="https://github.com/user-attachments/assets/da031f77-6b8d-4627-bff1-ad9a082d94b7" />

**Comment by KingDuckZ** [comment]
Any thoughts on my proposal from April? Would that be hard to implement while a better solution is not available?

To expand on it, given Reinachan's example, let's say I have Dr Mario with `grouping="Super Mario;Nintendo"` and F-Zero with `grouping="F-Zero;Nintendo", and both with `genre="Soundtrack"`.
What I'm proposing is an option to be entered in navidrome.toml like `RemapGenreToGrouping = true` for example, or `GenreTagName = 'grouping'`. When set like this, browsing through my collection on apps I would then expect to see these available genres: Super Mario, Nintendo, F-Zero. Then clicking on each I'd expect to see the relevant albums. Note that I wouldn't expect to see the "Soundtrack" genre anywhere at all.
I know it's hacky but it does enable browsing by grouping without any changes needed on the API side
